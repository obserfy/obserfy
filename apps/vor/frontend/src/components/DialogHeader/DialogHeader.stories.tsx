import { FC } from "react"
import { action } from "@storybook/addon-actions"
import DialogHeader from "./DialogHeader"

export default {
  title: "Core/DialogHeader",
  component: DialogHeader,
  parameters: {
    componentSubtitle: "Just a simple DialogHeader",
  },
}

export const Basic: FC = () => (
  <DialogHeader
    onAccept={action("accept")}
    title="A Dialog"
    onAcceptText="Save"
    onCancel={action("cancel")}
  />
)
