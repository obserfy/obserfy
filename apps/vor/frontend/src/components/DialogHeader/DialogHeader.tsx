import { Trans } from "@lingui/macro"
import { FC } from "react"
import { Button, Flex } from "theme-ui"
import LoadingIndicator from "../LoadingIndicator/LoadingIndicator"
import Typography from "../Typography/Typography"

interface Props {
  onCancel?: () => void
  onCancelText?: string
  onAccept?: () => void
  onAcceptText?: string
  title: string
  loading?: boolean
  disableAccept?: boolean
}
export const DialogHeader: FC<Props> = ({
  title,
  onAccept,
  onAcceptText = "Save",
  onCancel,
  onCancelText = "Cancel",
  loading,
  disableAccept,
}) => (
  <Flex
    backgroundColor="surface"
    py={1}
    sx={{
      alignItems: "center",
      position: "relative",
    }}
  >
    <Typography.H6
      sx={{
        fontSize: 2,
        width: "100%",
        position: "absolute",
        pointerEvents: "none",
        textAlign: "center",
        alignContent: "center",
      }}
    >
      <Trans id={title} />
    </Typography.H6>
    <Button
      variant="text"
      color="danger"
      my={1}
      onClick={onCancel}
      ml={2}
      data-cy="cancel"
    >
      <Trans id={onCancelText} />
    </Button>
    <Button
      variant="text"
      ml="auto"
      my={1}
      onClick={onAccept}
      mr={2}
      data-cy="confirm"
      disabled={disableAccept}
    >
      {loading && <LoadingIndicator />} <Trans id={onAcceptText} />
    </Button>
  </Flex>
)

export default DialogHeader
