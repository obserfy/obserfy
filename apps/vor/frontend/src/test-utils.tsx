/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
