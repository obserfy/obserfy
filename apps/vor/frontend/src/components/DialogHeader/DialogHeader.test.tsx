import { render } from "../../test-utils"
import { Basic } from "./DialogHeader.stories"

describe("DialogHeader", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
