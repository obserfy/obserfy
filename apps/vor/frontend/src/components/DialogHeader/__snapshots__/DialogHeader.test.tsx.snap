// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogHeader should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-top: 4px;
  padding-bottom: 4px;
  background-color: var(--theme-ui-colors-surface);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 20px;
  font-size: 16px;
  width: 100%;
  position: absolute;
  pointer-events: none;
  text-align: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
  margin-top: 4px;
  margin-bottom: 4px;
  margin-left: 8px;
  color: var(--theme-ui-colors-danger);
}

.emotion-3:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-3:hover,
.emotion-3:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
  margin-left: auto;
  margin-top: 4px;
  margin-bottom: 4px;
  margin-right: 8px;
}

.emotion-4:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-4:hover,
.emotion-4:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <h6
        class="emotion-2"
      >
        A Dialog
      </h6>
      <button
        class="emotion-3"
        data-cy="cancel"
      >
        Cancel
      </button>
      <button
        class="emotion-4"
        data-cy="confirm"
      >
         
        Save
      </button>
    </div>
  </div>
</div>
`;
